capacity/parse_capacity_change_request.js

```diff
@@ -83,6 +85,19 @@ module.exports = ({from, records}) => {
 
   const decrease = decodeNumber((decreaseRecord || defaultRecord).value);
   const increase = decodeNumber((increaseRecord || defaultRecord).value);
+  
+  let publicPrivate;
+  let newChannelType;
+  if(!!publicPrivateRecord) {
+    publicPrivate = decodeNumber(publicPrivateRecord.value);
```

Comment:
Instead of decoding a number it would be checking the bit on the channel flags byte `!(flagsNumber & 1)`

Reply:
its weird, its somehow not picking up if I use a bit for encoding, it doesn't find the record and decode it when the 2nd peer runs the command.

Reply:
I think maybe you are setting a record type not the record value? I can take a look at this in a bit when I pick this one up

Reply:
tried setting values to 0 and 1 as well. some how it completely skips parsing the change request. yeah, please let me know what you find when you get a chance to look at it. maybe I am missing something.

Reply:
@alexbosworth  I was able to point to what's causing it to fail. 

```
const publicRecordType = 0;
const privateRecordType = 1;

  if(!!new_channel_type) {
    records.push({
      type: new_channel_type === 'public' ? publicRecordType : privateRecordType,
      value: new_channel_type === 'public' ? Buffer.from([publicRecordType]) : Buffer.from([privateRecordType]),
    })
  }
  ```
  
P2P encoding fails to encode this extra piece and send it over to the peer. It's failing inside `encode_peer_request.js` file in the p2p folder, it never returns anything back. It does not throw errors though, it just returns blank.
I am not fully sure what data encodeTlvStream accepts.